llmfoundry/models/mpt/modeling_mpt.py

```diff
@@ -440,14 +442,187 @@ def construct_blocks(self, config: MPTConfig) -> nn.ModuleList:
             nn.ModuleList: The list of Transformer blocks.
         """
         block_args = self.extract_block_args(config.to_dict())
+        self.kv_cache_layers = None
+
+        if config.block_overrides is not None:
+            block_args_list = self._get_override_block_args_list(
+                config,
+                block_args,
+            )
+        else:
+            block_args_list = [block_args for _ in range(config.n_layers)]
 
         return nn.ModuleList([
             self.block_class(
                 device=config.init_device,
-                **block_args,
-            ) for _ in range(config.n_layers)
+                **block_args_i,
+            ) for block_args_i in block_args_list
         ])
 
+    def _get_override_block_args_list(
+        self,
+        config: MPTConfig,
+        block_args: Dict[str, Any],
+    ) -> List[Dict[str, Any]]:
+        if config.block_overrides is None:
+            raise ValueError(
+                'config.block_overrides should not be None when calling _get_override_block_args_list.',
+            )
+        repeat = config.block_overrides[
```

Comment:
should just `config.block_overrides.get('repeat', 1)

Reply:
Done.